File: CodingInterviewsPythonSolutions/240. Search a 2D Matrix II.py
class Solution:
    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:
        rows = len(matrix)
        cols = len(matrix[0])

        startRow = 0
        startCol = cols - 1

        while startRow < rows and startCol >= 0:
            if matrix[startRow][startCol] == target:
                return True
            elif matrix[startRow][startCol] < target:
                startRow += 1
            else:
                startCol -= 1

        return False

File: CodingInterviewsPackSolutionsCPP/240. Search a 2D Matrix II.cpp
class Solution {
public:
    bool searchMatrix(vector<vector<int>>& matrix, int target) {
        int rows = matrix.size();
        int cols = matrix[0].size();

        int startRow = 0;
        int startCol = cols-1;

        while(startRow<rows && startCol >=0){
            if(matrix[startRow][startCol]==target)
                return true;
            else if(matrix[startRow][startCol]<target){
                startRow++;
            }else{
                startCol--;
            }
        }
        return false;
    }
};